### PR TITLE
knowledge graph guides

### PR DESCRIPTION
**Pull request by mbana641**
Adding guides for the arcgis.graph module for working with knowledge graphs in the API for Python

I referenced other guides for formatting of the table of contents, please let me know if there's a better way to do that portion.

-----

# Checklist

Please go through each entry in the below checklist and mark an 'X' if that condition has been met. Every entry should be marked with an 'X' to be get the Pull Request approved.

- [x] All `import`s are in the first cell? 
    - [x] First block of imports are standard libraries
    - [x] Second block are 3rd party libraries
    - [x] Third block are all `arcgis` imports? Note that in some cases, for samples, it is a good idea to keep the imports next to where they are used, particularly for uncommonly used features that we want to highlight.
- [x] All `GIS` object instantiations are one of the following?
    - `gis = GIS()`
    - `gis = GIS('home')` or `gis = GIS('pro')`
    - `gis = GIS(profile="your_online_portal")`
    - `gis = GIS(profile="your_enterprise_portal")`
- [x] If this notebook requires setup or teardown, did you add the appropriate code to `./misc/setup.py` and/or `./misc/teardown.py`? (N/A)
- [x] If this notebook references any portal items that need to be staged on AGOL/Python API playground, did you coordinate with a Python API team member to stage the item the correct way with the `api_data_owner` user? (N/A)
- [x] If the notebook requires working with local data (such as CSV, FGDB, SHP, Raster files), upload the files as items to the [Geosaurus Online Org](geosaurus.maps.arcgis.com/) using `api_data_owner` account and change the notebook to first download and unpack the files. (N/A)
- [x] Code simplified & split out across multiple cells, useful comments?
- [x] Consistent voice/tense/narrative style? Thoroughly checked for typos?
- [x] All images used like `` instead of ``? All map widgets contain a static image preview? (Call `mapview_inst.take_screenshot()` to do so)
- [x] All file paths are constructed in an OS-agnostic fashion with `os.path.join()`? (Instead of `r"\foo\bar"`, `os.path.join(os.path.sep, "foo", "bar")`, etc.) (N/A)
- [x] Is your code formatted using [Jupyter Black](https://www.freecodecamp.org/news/auto-format-your-python-code-with-black/)? You can use Jupyter Black to format your code in the  notebook.
- [x] **IF YOU WANT THIS SAMPLE TO BE DISPLAYED ON THE DEVELOPERS.ARCGIS.COM WEBSITE**, ping @ mohi9282 so he can add it to the list for the next deploy

### CODE REVIEW

**Review by jyaistMap — APPROVED** [review]
Guides all read really well and look great when the developer site is built locally:

* screenshots: Intro
image
image

* Search
image
image

* Edit
image
image

**Comment by jyaistMap** [comment]
Thanks, @mbana641. Everything looks great!

**Review by nparavicini7 — APPROVED** [review]
Nice job @mbana641 !

**Comment by mbana641** [comment]
@jyaistMap @nparavicini7 I quickly wrote up some content for the guides for the new features we added this release, would it be alright for me to add those to this PR or would you rather I open a new one after this is merged? Thank you for reviewing!

**Comment by jyaistMap** [comment]
> @jyaistMap @nparavicini7 I quickly wrote up some content for the guides for the new features we added this release, would it be alright for me to add those to this PR or would you rather I open a new one after this is merged? Thank you for reviewing!

@mbana641 Yes, it's fine to add them to this PR.

File: guide/17-working-with-knowledge-graphs/part2_search_query_knowledge_graph.ipynb [hunk 1]
```diff
@@ -112,8 +113,110 @@
...
+    "        }\n",
+    "    ]\n",
+    "]\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
```
Comment (jyaistMap) [inline guide/17-working-with-knowledge-graphs/part2_search_query_knowledge_graph.ipynb:219]: * Reads and looks great when built locally:
image

File: guide/17-working-with-knowledge-graphs/part3_edit_knowledge_graph.ipynb [hunk 2]
```diff
@@ -383,6 +388,56 @@
...
+    "            prop_list.append(prop)\n",
+    "    knowledge_graph.update_search_index(adds={entity_type: {\"property_names\": prop_list}})\n",
+    "\n",
+    "# delete a property from the search index\n",
+    "knowledge_graph.update_search_index(deletes={\n",
+    "    \"User\": {\n",
+    "        \"property_names\": [\"name\"]\n",
+    "    }\n",
+    "})"
+   ]
```
Comment (jyaistMap) [inline guide/17-working-with-knowledge-graphs/part3_edit_knowledge_graph.ipynb:440]: Reads and looks great locally!
image

**Review by jyaistMap — APPROVED** [review]

File: guide/17-working-with-knowledge-graphs/part3_edit_knowledge_graph.ipynb (hunk 2 above)
Comment (nparavicini7) [reply guide/17-working-with-knowledge-graphs/part3_edit_knowledge_graph.ipynb:440]: small typo with search -> searched

**Comment by nparavicini7** [comment]
@mbana641 @jyaistMap new parts look great, good job! Just saw one typo, see my comment.

**Comment by jyaistMap** [comment]
Thank you @mbana641 - the guides should appear on the developer website the 27th or 28th

**Pull request merged**